src/components/InstallNitric.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useEffect, useState } from 'react'
+import { CodeGroup } from './Code'
+
+// this function detects the operating system of the user and returns the name that will be used to sort the instructions
+function detectOS() {
+  const userAgent = window.navigator.userAgent
```

Comment:
Can navigator ever be undefined/null?

Reply:
it can be, however it is only called on client using useEffect